### PR TITLE
Fix debug build crash by out of range assert

### PR DESCRIPTION
**Pull request by Sanhaoji2**

### CODE REVIEW

**Review by harsha-simhadri — APPROVED** [review]

**Comment by shanewil** [comment]
Note, this problem will get fixed by pr #163. It's currently in review and requires some small changes by the author.

**Comment by shanewil** [comment]
Note, this issue will get fixed by pr #163. It's currently in review and waiting on some small changes before approval.

**Pull request closed without merging**
